### PR TITLE
Issue462 modular hydraulic systems admix

### PR DESCRIPTION
**Pull request by alexanderAKU**
> Thank you for making a Pull Request to AixLib!

> Please consider these points before submitting:
> - Mention the issue that this Pull Request intends to solve, and briefly describe what the main changes are.
> - Make sure all new code follows our [Modelica Guidelines](https://github.com/RWTH-EBC/AixLib/wiki/Modelica-guidelines)
> - Make sure all new models include documentation as described in the [Documentation Wiki-Page](https://github.com/RWTH-EBC/AixLib/wiki/Documentation) and are demonstrated by suitable examples

> If you know someone who could review your code, please use the *Reviewer* function on the right to ask for a review and also assign the PR to them. If not, no worries, we will find someone for you.

> Thanks again for your PR! You can delete these first lines before submitting your PR.

### CODE REVIEW

**Comment by mlauster** [comment]
@alexanderAKU, can you please resolve merge conflicts? @PMehrfeld, can you please review?

File: AixLib/Fluid/HydraulicModules/HydraulicBus.mo [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  SI.Temperature TrtrnIn "Temperature into return line";
+  SI.Temperature TrtrnOut "Temperature out of return line";
+  annotation (
+    Icon(graphics,
+         coordinateSystem(preserveAspectRatio=false)),
+    Diagram(graphics,
+            coordinateSystem(preserveAspectRatio=false)),
+    Documentation(info="<html>
+<p>
+Definition of a standard module bus for use with the Zugabe library. A module
```
Comment (PMehrfeld) [inline AixLib/Fluid/HydraulicModules/HydraulicBus.mo]: Revise documentation such that Zugabe is not mentioned anymore.

File: AixLib/Fluid/HydraulicModules/HydraulicBus.mo [hunk 2]
```diff
@@ -0,0 +1,31 @@
+within AixLib.Fluid.HydraulicModules;
+expandable connector HydraulicBus "data bus for hydraulic circuits"
+  extends Modelica.Icons.SignalBus;
+  import SI = Modelica.SIunits;
+  Real rpm_Input "pump speed"
+    annotation (HideResult=false);
+  SI.Power P "electrical pump power" annotation (HideResult=false);
+  Real valveSet "Opening valve 0-1";
```
Comment (PMehrfeld) [inline AixLib/Fluid/HydraulicModules/HydraulicBus.mo]: min, max as attributes.

File: AixLib/Fluid/HydraulicModules/HydraulicBus.mo [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  SI.Temperature Tambient "Ambient temperature";
```
Comment (PMehrfeld) [inline AixLib/Fluid/HydraulicModules/HydraulicBus.mo]: T_amb
according to https://github.com/RWTH-EBC/AixLib/wiki/Variable-Naming

File: AixLib/Fluid/HydraulicModules/HydraulicBus.mo [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+  SI.Temperature TfwrdIn "Flow Temperature into forward line";
```
Comment (PMehrfeld) [inline AixLib/Fluid/HydraulicModules/HydraulicBus.mo]: TFwrdIn or TFwrd_in
Also see following 3 variables

File: AixLib/Fluid/HydraulicModules/Example/test_Admix.mo [hunk 5]
```diff
@@ -0,0 +1,95 @@
+within AixLib.Fluid.HydraulicModules.Example;
+model test_Admix "test for admix circuit"
```
Comment (PMehrfeld) [inline AixLib/Fluid/HydraulicModules/Example/test_Admix.mo]: just `Admix` as class name.

File: AixLib/Fluid/HydraulicModules/HydraulicBus.mo [hunk 6]
```diff
@@ -0,0 +1,31 @@
+within AixLib.Fluid.HydraulicModules;
+expandable connector HydraulicBus "Data bus for hydraulic circuits"
+  extends Modelica.Icons.SignalBus;
+  import SI = Modelica.SIunits;
+  Real rpm_Input "Pump speed"
```
Comment (PMehrfeld) [inline AixLib/Fluid/HydraulicModules/HydraulicBus.mo]: Maybe use Modelica.SIunits.Conversions.NonSIunits.AngularVelocity_rpm?!

File: AixLib/Fluid/HydraulicModules/HydraulicBus.mo [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+    annotation (HideResult=false);
+  SI.Power P "Electrical pump power" annotation (HideResult=false);
+  Real valveSet "Opening valve 0..1";
```
Comment (PMehrfeld) [inline AixLib/Fluid/HydraulicModules/HydraulicBus.mo]: min, max as attributes.

File: AixLib/Fluid/HydraulicModules/HydraulicBus.mo [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+  SI.Temperature Tambient "Ambient temperature";
```
Comment (PMehrfeld) [inline AixLib/Fluid/HydraulicModules/HydraulicBus.mo]: T_amb
according to https://github.com/RWTH-EBC/AixLib/wiki/Variable-Naming

File: AixLib/Fluid/HydraulicModules/HydraulicBus.mo [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+  SI.Temperature TfwrdIn "Flow Temperature into forward line";
```
Comment (PMehrfeld) [inline AixLib/Fluid/HydraulicModules/HydraulicBus.mo]: TFwrdIn or TFwrd_in
Also see following 3 variables

**Comment by PMehrfeld** [comment]
grafik

Parameters should be grouped via annotations.

File: AixLib/Fluid/HydraulicModules/Admix.mo [hunk 10]
```diff
@@ -0,0 +1,464 @@
...
+        system.p_start,
+        Tinit,
+        Medium.X_default),
+    tau=0.2)                           annotation (Dialog(enable=true),
+      Placement(transformation(
+        extent={{-10,-10},{10,10}},
+        rotation=0,
+        origin={-30,20})));
+
+public
```
Comment (PMehrfeld) [inline AixLib/Fluid/HydraulicModules/Admix.mo]: make a public section and below a protected one (if really necessary)

File: AixLib/Fluid/HydraulicModules/Admix.mo [hunk 11]
```diff
@@ -0,0 +1,464 @@
...
+
+  replaceable package Medium =
+      Modelica.Media.Water.ConstantPropertyLiquidWater
+    constrainedby Modelica.Media.Interfaces.PartialMedium
+    "Medium in the system" annotation (choicesAllMatching=true);
+  AixLib.Fluid.Actuators.Valves.ThreeWayEqualPercentageLinear val(
+    CvData=AixLib.Fluid.Types.CvTypes.Kv,
+    l={0.001,0.001},
+    Kv=10.0,
+    m_flow_nominal=8*996/3600,
```
Comment (PMehrfeld) [inline AixLib/Fluid/HydraulicModules/Admix.mo]: Are you sure with all these set parameters? 
Better leave out as long as the parameter values are not 100 % generic.

File: AixLib/Fluid/HydraulicModules/Admix.mo [hunk 12]
```diff
@@ -0,0 +1,464 @@
+within AixLib.Fluid.HydraulicModules;
+model Admix "Admix circuit with three way valve and rpm controlled pump"
+
+  parameter Modelica.SIunits.Volume vol=0.0005 "Mixing Volume";
+  parameter Modelica.SIunits.Temperature Tinit=303.15
+    "Initialization temperature";
+  parameter Modelica.SIunits.Time tau=15
+    "Time Constant for PT1 behavior of temperature sensors";
+
+  replaceable package Medium =
```
Comment (PMehrfeld) [inline AixLib/Fluid/HydraulicModules/Admix.mo]: Just use `  replaceable package Medium2 = Modelica.Media.Interfaces.PartialMedium`
Or even much better extend `AixLib.Fluid.Interfaces.PartialFourPort` or `PartialFourPortInterface`

File: AixLib/Fluid/HydraulicModules/Admix.mo [hunk 13]
```diff
@@ -0,0 +1,464 @@
...
+      Modelica.Media.Water.ConstantPropertyLiquidWater
+    constrainedby Modelica.Media.Interfaces.PartialMedium
+    "Medium in the system" annotation (choicesAllMatching=true);
+  AixLib.Fluid.Actuators.Valves.ThreeWayEqualPercentageLinear val(
+    CvData=AixLib.Fluid.Types.CvTypes.Kv,
+    l={0.001,0.001},
+    Kv=10.0,
+    m_flow_nominal=8*996/3600,
+    dpFixed_nominal={8000,8000},
+    redeclare package Medium = Medium,
```
Comment (PMehrfeld) [inline AixLib/Fluid/HydraulicModules/Admix.mo]: Think of using `final` statements for parametrization. E.g. `redeclare final package Medium = Medium,`.

File: AixLib/Fluid/HydraulicModules/Admix.mo [hunk 14]
```diff
@@ -0,0 +1,464 @@
+within AixLib.Fluid.HydraulicModules;
+model Admix "Admix circuit with three way valve and rpm controlled pump"
+
+  parameter Modelica.SIunits.Volume vol=0.0005 "Mixing Volume";
+  parameter Modelica.SIunits.Temperature Tinit=303.15
```
Comment (PMehrfeld) [inline AixLib/Fluid/HydraulicModules/Admix.mo]: T_start is more common. However, this parameter would come along with `AixLib.Fluid.Interfaces.LumpedVolumeDeclarations` (see the comment regarding `ClosedVolume`)

File: AixLib/Fluid/HydraulicModules/Admix.mo [hunk 15]
```diff
@@ -0,0 +1,464 @@
...
+            50},{110,70}})));
+  Modelica.Fluid.Interfaces.FluidPort_a port_a2(redeclare package Medium =
+        Medium) "Return line inflowing medium"
+    annotation (Placement(transformation(extent={{90,-70},{110,-50}})));
+  HydraulicModules.HydraulicBus hydraulicBus
+    annotation (Placement(transformation(extent={{-20,80},{20,120}}),
+        iconTransformation(extent={{-20,80},{20,120}})));
+  FixedResistances.Pipe               pipe1(redeclare package Medium = Medium,
+      T_start=Tinit,
+    parameterPipe=DataBase.Pipes.Copper.Copper_35x1_5())
```
Comment (PMehrfeld) [inline AixLib/Fluid/HydraulicModules/Admix.mo]: Is it really necessary to choose a pipe record in this class? Preferably, change the records in the example model. (Same for all pipes and other components.)

File: AixLib/Fluid/HydraulicModules/Admix.mo [hunk 16]
```diff
@@ -0,0 +1,464 @@
...
+  FixedResistances.Pipe                                  pipe4(redeclare
+      package Medium =                                                                    Medium,
+      T_start=Tinit,
+    parameterPipe=DataBase.Pipes.Copper.Copper_35x1_5())
+                     annotation (Dialog(enable=true), Placement(transformation(
+        extent={{-6,6},{6,-6}},
+        rotation=180,
+        origin={46,-60})));
+protected
+  Modelica.Fluid.Vessels.ClosedVolume junc456(
```
Comment (PMehrfeld) [inline AixLib/Fluid/HydraulicModules/Admix.mo]: Might `AixLib.Fluid.MixingVolumes.MixingVolume` the better class? Examplarily, see `AixLib.Airflow.Multizone.MediumColumnDynamic` for passing through all parameters.
Especially this means:
`extends AixLib.Fluid.Interfaces.LumpedVolumeDeclarations`
and using `final` statements.

File: AixLib/Fluid/HydraulicModules/Admix.mo [hunk 17]
```diff
@@ -0,0 +1,464 @@
...
+                     annotation (Dialog(enable=true), Placement(transformation(
+        extent={{-6,6},{6,-6}},
+        rotation=90,
+        origin={-30,-22})));
+
+  // -------------------------------------------------
+  // Sensors
+  // -------------------------------------------------
+protected
+  Modelica.Fluid.Sensors.VolumeFlowRate vFRfwrdIn(redeclare package Medium =
```
Comment (PMehrfeld) [inline AixLib/Fluid/HydraulicModules/Admix.mo]: Using AixLib / IBPSA Sensors is preferred.

How about naming the component: VFRSenFlow_in
At least the "F" must be a capital letter in order to stick to CamelCase

File: AixLib/Fluid/HydraulicModules/Admix.mo [hunk 18]
```diff
@@ -0,0 +1,464 @@
...
+        Medium) "inflow into admix module in forward line" annotation (
```
Comment (PMehrfeld) [inline AixLib/Fluid/HydraulicModules/Admix.mo]: Start comments with a capital letter (all component comments)

**Comment by PMehrfeld** [comment]
@alexanderAKU: Thanks a lot for this admix module!
Generally, it looks quite good. However, it would be more beneficious to extend from and stick to IBPSA base models and rules.

Regarding the icon: It looks really nice. Anyway, I suggest to expand the lower dashed line to the lower icon boundary. Furthermore, adding strings for pipe1..pipe6 would be beneficious.

File: AixLib/Fluid/HydraulicModules/Admix.mo (hunk 16 above)
Comment (alexanderAKU) [reply AixLib/Fluid/HydraulicModules/Admix.mo]: @PMehrfeld why is the mixingVolume the better class?

Comment (PMehrfeld) [reply AixLib/Fluid/HydraulicModules/Admix.mo]: @alexanderAKU : This was just meant as a hint to consider this class. It is not necessarily a better choice.

**Comment by PMehrfeld** [comment]
This branch is now in `issue462_ModularHydraulicSystems` which is the general branch that will be merged into development at some point.
Thus, I will close this PR and delete the branch `issue462_ModularHydraulicSystems_admix`.

**Pull request closed without merging**